src/main/java/mate/academy/bookshop/dto/order/OrderResponseDto.java

```diff
@@ -0,0 +1,26 @@
+package mate.academy.bookshop.dto.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Data;
+import mate.academy.bookshop.dto.user.UserResponseDto;
+import mate.academy.bookshop.model.Status;
+
+@Data
+public class OrderResponseDto implements Comparable<OrderResponseDto> {
+    private Long id;
+    private UserResponseDto user;
+    private Set<OrderItemResponseDto> orderItems = new HashSet<>();
+    private Status status;
+    private BigDecimal total;
+    private LocalDateTime orderDate;
+    private String shippingAddress;
+
+    @Override
+    public int compareTo(OrderResponseDto dto) {
+        return orderDate.isBefore(dto.orderDate) ? -1 :
+        orderDate.isAfter(dto.orderDate) ? 1 : 0;
+    }

```

Comment:
I'd avoid something like this and used repository sorting instead, could be also sorted by id, just define some default `Sort` in controller and pass it down to the repository method. And I think pageable also can contain sorting info inside it, but you gonna have to double check